js/spreadsheet.js

```diff
@@ -36,31 +33,30 @@ var spreadsheet = function(options) {
    * @param {Array} columns
    * @param {Boolean} isFirstRender - defines if it was first render to prepare the data for correct rendering without data changes after changes are made
    */
+
   function prepareData(rows, columns, isFirstRender) {
     var preparedData = rows.map(function(row) {
-      var dataRow = columns.map(function(header, index) {
+      var dataRow = columns.map(function(header) {
         var value = row.data[header];
 
-        if (Array.isArray(value)) {
-          if (arrayColumns.indexOf(header) === -1) {
-            arrayColumns.push(header);
-          }
-
-          // Add double quotes to the string if it contains a comma
-          value = value.map(function(val) {
-            // Stringify value only for the first render for nested arrays
-            if (isFirstRender && value && typeof val !== 'string') {
-              return JSON.stringify(val);
-            }
-
-            return typeof val === 'string' && val.indexOf(',') !== -1 ? '"' + val + '"' : val;
-          }).join(', ');
         // Stringify value only for the first render for nested objects
-        } else if (isFirstRender && value && typeof value === 'object') {
+        if (isFirstRender && typeof value === 'object') {
           if (objColumns.indexOf(header) === -1) {
             objColumns.push(header);
           }
 
+          if (Array.isArray(value)) {
```

Comment:
@OleksandrLavrynenko I'm reading this section and there are a few things that don't make sense to me.

**Objective**

- Stringify objects/arrays so that they are representative of the content, e.g. arrays of objects use `[]` as the wrapper as expected, objects aren't displayed as `[object Object]` etc.
- Objects/arrays are saved correctly

I therefore see a few problems here:

1. Why do we need the logic to add double quotes to the string if it contains a comma?
2. Why do we have a special scenario for nested arrays?

Could you also share screenshots of what this looks like if the data was the following:

* `[1, "2, and 3", 4, { foo: "bar" }]`
* `[1, 2, [3, 4, "hi, how are you"], 5, 6, { foo: ["bar", "buzz"] }]

Reply:
**@tonytlwu**  There is a problem with parsing the first line

![compare-dsta](https://user-images.githubusercontent.com/86256215/141450062-12805c0d-6d79-4a3b-84ad-acfdc69e204e.jpg)

Reply:
@OleksandrLavrynenko I also don't see `[1, 2, [3, 4, "hi, how are you"], 5, 6, { foo: ["bar", "buzz"] }]` being rendered correctly. Can you fix both issues please?

I expect the following:

* `[1, "2, and 3", 4, { foo: "bar" }]` => `[1,"2, and 3",4,{"foo":"bar"}]`
* `[1, 2, [3, 4, "hi, how are you"], 5, 6, { foo: ["bar", "buzz"] }]` => `[1,2,[3,4,"hi, how are you"],5,6,{"foo":["bar","buzz"]}]`

Reply:
@OleksandrLavrynenko Please show screenshot.